core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -336,6 +340,70 @@ public LogicalPlan visitAggregation(Aggregation node, AnalysisContext context) {
     return new LogicalAggregation(child, aggregators, groupBys);
   }
 
+  @Override
+  public LogicalPlan visitFieldSummary(FieldSummary node, AnalysisContext context) {
+    LogicalPlan child = node.getChild().getFirst().accept(this, context);
+
+    TypeEnvironment env = context.peek();
+    Map<String, ExprType> fieldsMap = env.lookupAllFields(Namespace.FIELD_NAME);
+
+    if (node.getIncludeFields() != null) {
+      List<String> includeFields =
+          node.getIncludeFields().stream()
+              .map(expr -> ((Field) expr).getField().toString())
+              .toList();
```

Comment:
you can do your filter and collect to map in your stream, and then return a map.